kiwi_stackbuild_plugin/xml_merge.py

```diff
@@ -0,0 +1,333 @@
+# Copyright (c) 2021 SUSE Linux GmbH.  All rights reserved.
+#
+# This file is part of kiwi-stackbuild.
+#
+# kiwi is free software: you can redistribute it and/or modify
+# it under the terms owf the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# kiwi is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with kiwi.  If not, see <http://www.gnu.org/licenses/>
+#
+from lxml import etree
+import os
+import glob
+import logging
+
+from kiwi.utils.sync import DataSync
+from kiwi.defaults import Defaults
+from kiwi.exceptions import (
+    KiwiConfigFileNotFound
+)
+
+from kiwi_stackbuild_plugin.defaults import StackBuildDefaults
+from kiwi_stackbuild_plugin.exceptions import (
+    KiwiStackBuildPluginSchemaValidationFailed
+)
+
+log = logging.getLogger('kiwi')
+
+
+class XMLMerge:
+    """
+    ***Implements the class to handle stackbuild schema and descriptions.
+    Converts a given stackbuild description into a full KIWI description
+    based on the stashed KIWI description***
+
+    :param str description_dir: the folder path contianing the stackbuild
+        description XML
+    """
+    def __init__(self, description_dir: str):
+        self.description = self._find_description_file(description_dir)
+        self.description_dir = description_dir
+        self.description_tree = etree.parse(self.description)
+
+    def is_stackbuild_description(self) -> bool:
+        """
+        Returns True if the root element includes the 'stackbuild="true"'
+        attribute
+
+        :returns: True if the stackbuild attribute is found, False otherwise
+
+        :rtype: bool
+        """
+        root = self.description_tree.getroot()
+        if 'stackbuild' in root.attrib and root.attrib['stackbuild'] == 'true':
+            return True
+        return False
+
+    def validate_schema(self) -> None:
```

Comment:
This is the logic that I could not manage to figure out at a time, how to include the KIWI schema file during validation being sure the schema I am including is the one that will be actually in use by currently loaded module (there could be more than one KIWI installation around), so basically it is not an option to hardcode any path in the `schema.rnc` as I could not figure out a relative path that works for RPM installation, pip install, virtualenvs, etc.
The work around is to dynamically set the KIWI schema path at run time with the `Defaults.get_schema_file` method. Then use a dummy fake schema to run the convertion from `rnc` to `rng`, because `trang` also converts and parses any included file...